### PR TITLE
return cpu time for knitro_version >= 12

### PR DESCRIPTION
**Pull request by tmigot**
- Add a constant for Knitro's version as in [Knitro.jl (L32)](https://github.com/jump-dev/KNITRO.jl/blob/12da5707db3195f915f7f7dd7f9cb051414c2c4b/src/KNITRO.jl#L32).
- If `KNITRO_VERSION ≥ v"12.0"` we return the CPU time computed by the solver instead of the result of `@timed`.

Note that Knitro also returns the "real" time with `KNITRO.KN_get_solve_time_real(kc)`, see the functions in [KNITRO.jl (L352)](https://github.com/jump-dev/KNITRO.jl/blob/12da5707db3195f915f7f7dd7f9cb051414c2c4b/src/kn_attributes.jl#L352).

### CODE REVIEW

File: src/NLPModelsKnitro.jl [hunk 1]
```diff
@@ -4,6 +4,8 @@ export knitro
 
 using NLPModels, KNITRO, SolverTools
 
+const KNITRO_VERSION = KNITRO.unsafe_get_release() #Require KNITRO_VERSION ≥ v"11.0"
```
Comment (dpo) [inline src/NLPModelsKnitro.jl:7]:
```suggestion
const KNITRO_VERSION = KNITRO.unsafe_get_release()
```

**Review by dpo — COMMENTED** [review]
Was this comment needed?

**Comment by dpo** [comment]
Maybe it's the real time that makes sense, as it also contains the time to call the interface and to call Knitro itself?

**Comment by tmigot** [comment]
> Maybe it's the real time that makes sense, as it also contains the time to call the interface and to call Knitro itself?

On principle, I agree. I suggested first the CPU time to be consistent with [NLPModelsIpopt.jl  L138](https://github.com/JuliaSmoothOptimizers/NLPModelsIpopt.jl/blob/991955d7bd4362f8b9dd76a48d94733d368c0c6d/src/NLPModelsIpopt.jl#L138) which also seems to return this information.

**Comment by tmigot** [comment]
In the first version, I didn't realize the file contains two calls to the `knitro` functions.
It is good now.

File: src/NLPModelsKnitro.jl (hunk 1 above)
Comment (tmigot) [reply src/NLPModelsKnitro.jl]: removed

**Comment by tmigot** [comment]
followed [PR 52 in NLPModelsIpopt.jl](https://github.com/JuliaSmoothOptimizers/NLPModelsIpopt.jl/pull/52).

**Comment by dpo** [comment]
Thank you. Let's go with this for now. I haven't found a Julia function that returns CPU time. So I wonder if the standard stats should always report real time, and the solver specific field should be reserved for CPU time, and will only be available when a low-level language library reports it.

**Pull request merged**